src/snowflake/cli/plugins/spcs/image_repository/manager.py

```diff
@@ -15,38 +21,42 @@ def get_schema(self):
     def get_role(self):
         return self._conn.role
 
-    def get_repository_url_list(self, repo_name: str) -> SnowflakeCursor:
-        role = self.get_role()
-        database = self.get_database()
-        schema = self.get_schema()
-
-        registry_query = f"""
-            use role {role};
-            use database {database};
-            use schema {schema};
-            show image repositories like '{repo_name}';
-            """
-
-        return self._execute_query(registry_query)
+    def get_repository_row(self, repo_name: str) -> Dict:
+        if not is_valid_unquoted_identifier(repo_name):
+            raise ValueError(
+                f"repo_name '{repo_name}' is not a valid unquoted Snowflake identifier"
+            )
 
-    def get_repository_url(self, repo_name):
-        database = self.get_database()
-        schema = self.get_schema()
+        repo_name = repo_name.upper()
 
-        result_set = self.get_repository_url_list(repo_name=repo_name)
+        # because image repositories only support unquoted identifiers, SHOW LIKE should only return one or zero rows
+        repository_list_query = (
+            f"show image repositories like '{escape_like_pattern(repo_name)}'"
```

Comment:
> If you use the backslash as an escape character, then you must specify escape the backslash in the ESCAPE clause.

https://docs.snowflake.com/en/sql-reference/functions/like#usage-notes Looks like we need to add `ESCAPE '\\';`

Reply:
it's a weird inconsistency but that only applies to LIKE with string matching. SHOW ... LIKE ... uses a different semantic with \\ as an escape character by default and also case-insensitive while LIKE is case-sensitive

https://docs.snowflake.com/en/sql-reference/sql/show-image-repositories